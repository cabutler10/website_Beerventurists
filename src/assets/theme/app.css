html {
  height: 100%;
}

body {
  margin: 0;
  padding: 0;
  font-family: 'Open Sans', sans-serif;
  background-color: #ffffff;
  overflow-x: hidden;
  font-size: 14px;
  width: '100%';
}

a {
  text-decoration: none;
}
