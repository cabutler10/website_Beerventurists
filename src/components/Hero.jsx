import React from 'react'
import PropTypes from 'prop-types'
import Img from 'gatsby-image'
import { useTranslation } from 'react-i18next'
import classnames from 'classnames'
import { makeStyles } from '@material-ui/styles'
import Typography from '@material-ui/core/Typography'
import Hidden from '@material-ui/core/Hidden'

const useStyles = makeStyles(theme => ({
  root: {
    height: 250,
    marginTop: -74,
    [theme.breakpoints.up('sm')]: {
      height: 300,
    },
    [theme.breakpoints.up('md')]: {
      height: 700,
    },
    [theme.breakpoints.up('xl')]: {
      height: 1000,
    },
  },
  container: {
    paddingTop: 100,
    zIndex: -1,
    margin: '0 auto',
    width: '100%',
    [theme.breakpoints.up('sm')]: {
      position: 'absolute',
      paddingTop: 100,
    },
    [theme.breakpoints.up('md')]: {
      paddingTop: 150,
    },
    [theme.breakpoints.up('lg')]: {
      paddingTop: 150,
    },
    [theme.breakpoints.up('xl')]: {
      paddingTop: 300,
    },
  },
  img: {
    zIndex: -5,
    left: 0,
    top: 0,
    width: '100%',
    height: 250,
    [theme.breakpoints.up('sm')]: {
      height: 300,
    },
    [theme.breakpoints.up('md')]: {
      height: 700,
    },
    [theme.breakpoints.up('xl')]: {
      height: 1000,
    },
  },
  text: {
    letterSpacing: 3,
    color: theme.status.black,
    textShadow: '2px 2px 4px #ccc',
  },
  title: {
    textTransform: 'capitalize',
    textAlign: 'center',
    fontSize: 35,
    fontWeight: 600,
    margin: 0,
    [theme.breakpoints.up('sm')]: {
      fontSize: 48,
    },
    [theme.breakpoints.up('md')]: {
      fontSize: 96,
    },
    [theme.breakpoints.up('lg')]: {
      fontSize: 122,
    },
  },
  small: {
    fontSize: 35,
    [theme.breakpoints.up('sm')]: {
      fontSize: 48,
    },
    [theme.breakpoints.up('md')]: {
      fontSize: 72,
    },
    [theme.breakpoints.up('lg')]: {
      fontSize: 96,
    },
  },
}))

function Hero({ img }) {
  const classes = useStyles()
  return (
    <div className={classes.root}>
      <Img
        fluid={img.node.fluid}
        alt="wave"
        className={classes.img}
        imgStyle={{ objectPosition: 'center top' }}
        style={{ position: 'absolute' }}
      />
      <div className={classes.container}>
        <Hidden xsDown>
          <Typography
            variant="h2"
            gutterBottom
            className={classnames(classes.text, classes.title)}
          >
            B<span className={classes.small}>EER</span>V
            <span className={classes.small}>ENTURISTS</span>
          </Typography>
        </Hidden>
        <Hidden smUp>
          <Typography
            variant="h2"
            gutterBottom
            className={classnames(classes.text, classes.title)}
          >
            B<span className={classes.small}>EER</span>V
            <span className={classes.small}>ENTURISTS</span>
          </Typography>
        </Hidden>
      </div>
    </div>
  )
}

Hero.propTypes = {
  img: PropTypes.object.isRequired,
}

export default Hero
