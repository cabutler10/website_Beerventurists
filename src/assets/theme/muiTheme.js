import { createMuiTheme } from '@material-ui/core/styles'
import 'typeface-open-sans'
import 'typeface-permanent-marker'
import './app.css'

const status = {
  white: '#ffffff',
  black: '#040001',
  greyLt: '#F7F5F5',
  greyMed: '#656565',
  grey: '#c5c6c7',
  greyDk: '#404040',
  green: '#00942b',
  yellow: '#fce566',
  red: '#FF0012',
  blue: '#015b99',
  blueLt: '#0172c0',
  blueDk: '#014472',
  orange: '#c56113',
  orangeDk: '#c56113',
  orangeLt: '#ebab2f',
  tan: '#FFFBEF',
}

const theme = createMuiTheme({
  palette: {
    type: 'light',
    primary: {
      light: status.blueLt,
      main: status.blue,
      dark: status.blueDk,
    },
    secondary: {
      light: status.orangeLt,
      main: status.orange,
      dark: status.orangeDk,
    },
    error: { main: status.red },
  },
  status,
  typography: {
    useNextVariants: true,
    fontFamily: "'Open Sans', sans-serif",
    h5: {
      letterSpacing: 3,
    },
    h2: {
      fontFamily: "'Permanent Marker', cursive",
    },
    body1: {
      letterSpacing: 1,
      lineHeight: 1.8,
      //TODO is this an error in MUI? not sure why its default is 16 when MUI default is 14
      fontSize: 14,
    },
    body2: {
      letterSpacing: 1,
      fontSize: 14,
    },
    button: {
      letterSpacing: 3,
    },
  },
  overrides: {
    MuiTypography: {
      colorPrimary: {
        color: status.black,
      },
      colorSecondary: {
        color: status.orangeLt,
      },
      colorTextSecondary: {
        color: status.white,
      },
    },
    MuiButton: {
      root: {
        padding: '9px 27px',
        borderRadius: 0,
      },
      contained: {
        boxShadow: 'none',
      },
    },
  },
})

export default theme
