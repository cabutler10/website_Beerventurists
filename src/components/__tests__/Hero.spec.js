import React from 'react'
import { render } from 'react-testing-library'
import Hero from '../Hero'
import { ThemeProvider } from '@material-ui/styles'
import theme from '../../assets/theme/muiTheme'

describe('Hero', () => {
  it('renders correctly', () => {
    const mockProps = {
      img: {
        node: {
          fluid: {
            aspectRatio: 0,
            src: 'mockString',
            srcSet: 'mockString',
            sizes: 'mockString',
          },
        },
      },
      t: () => '',
    }
    const component = render(
      <ThemeProvider theme={theme}>
        <Hero {...mockProps} />
      </ThemeProvider>
    )
    expect(component).toMatchSnapshot()
  })
})
