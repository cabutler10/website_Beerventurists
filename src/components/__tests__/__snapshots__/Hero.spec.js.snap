// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="Hook-root-1ko8ali"
      >
        <div
          class="Hook-img-fsctee gatsby-image-wrapper"
          style="position: absolute; overflow: hidden;"
        >
          <div
            style="width: 100%;"
          />
          <picture>
            <img
              alt="wave"
              sizes="mockString"
              src="mockString"
              srcset="mockString"
              style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; opacity: 0;"
            />
          </picture>
          <noscript>
            &lt;picture&gt;&lt;img sizes="mockString" srcset="mockString" src="mockString" alt="wave" style="position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center"/&gt;&lt;/picture&gt;
          </noscript>
        </div>
        <div
          class="Hook-container-jdq4g6"
        >
          <h2
            class="MuiTypography-root-1 MuiTypography-h2-14 MuiTypography-gutterBottom-28 Hook-text-n934yh Hook-title-1hy883f"
          >
            E
            <span
              class="Hook-small-d5kd93"
            >
              ISBACH
            </span>
          </h2>
          <h2
            class="MuiTypography-root-1 MuiTypography-h2-14 MuiTypography-gutterBottom-28 Hook-text-n934yh Hook-title-1hy883f Hook-title2-z9aol5"
          >
            R
            <span
              class="Hook-small-d5kd93"
            >
              IDERS
            </span>
          </h2>
          <p
            class="MuiTypography-root-1 MuiTypography-body1-10 Hook-text-n934yh Hook-subtitle-mygehx"
          >
            hero.subtitle
          </p>
          <div
            class="Hook-containerNewsletter-1aq479i"
          >
            <button
              class="MuiButtonBase-root-70 MuiButton-root-44 MuiButton-contained-55 MuiButton-containedPrimary-56 MuiButton-raised-58 MuiButton-raisedPrimary-59 MuiButton-fullWidth-69 Hook-button-1j9lk2z"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label-45"
              >
                newsletter.signUp
              </span>
              <span
                class="MuiTouchRipple-root-89"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="Hook-root-1ko8ali"
    >
      <div
        class="Hook-img-fsctee gatsby-image-wrapper"
        style="position: absolute; overflow: hidden;"
      >
        <div
          style="width: 100%;"
        />
        <picture>
          <img
            alt="wave"
            sizes="mockString"
            src="mockString"
            srcset="mockString"
            style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; opacity: 0;"
          />
        </picture>
        <noscript>
          &lt;picture&gt;&lt;img sizes="mockString" srcset="mockString" src="mockString" alt="wave" style="position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center"/&gt;&lt;/picture&gt;
        </noscript>
      </div>
      <div
        class="Hook-container-jdq4g6"
      >
        <h2
          class="MuiTypography-root-1 MuiTypography-h2-14 MuiTypography-gutterBottom-28 Hook-text-n934yh Hook-title-1hy883f"
        >
          E
          <span
            class="Hook-small-d5kd93"
          >
            ISBACH
          </span>
        </h2>
        <h2
          class="MuiTypography-root-1 MuiTypography-h2-14 MuiTypography-gutterBottom-28 Hook-text-n934yh Hook-title-1hy883f Hook-title2-z9aol5"
        >
          R
          <span
            class="Hook-small-d5kd93"
          >
            IDERS
          </span>
        </h2>
        <p
          class="MuiTypography-root-1 MuiTypography-body1-10 Hook-text-n934yh Hook-subtitle-mygehx"
        >
          hero.subtitle
        </p>
        <div
          class="Hook-containerNewsletter-1aq479i"
        >
          <button
            class="MuiButtonBase-root-70 MuiButton-root-44 MuiButton-contained-55 MuiButton-containedPrimary-56 MuiButton-raised-58 MuiButton-raisedPrimary-59 MuiButton-fullWidth-69 Hook-button-1j9lk2z"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label-45"
            >
              newsletter.signUp
            </span>
            <span
              class="MuiTouchRipple-root-89"
            />
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
